cloud/aws/providerconfig/types.go

```diff
@@ -14,22 +14,156 @@
 package providerconfig
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// AWSMachineProviderConfig is the type that will be embedded in a Machine.Spec.ProviderConfig field
+// for an AWS instance.
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 type AWSMachineProviderConfig struct {
```

Comment:
Is this supposed to represent a single machine? What about ASGs?

Reply:
@errordeveloper single machine - we won't use ASG

Reply:
I though this subproject wasn't going to be so prescriptive, and it would be up to any given flavour to make a decision.

Reply:
@errordeveloper That is a goal for the overall project, yes. That said, I think there is an inherent impedance mismatch between a Machine and an ASG-managed instance that we probably need to solve more generically in the great cluster-api.

To me it feels like the MachineSet controller would be a better manager for ASG-backed instances rather than the Machine controller. I am interested on hearing more of your thoughts on this topic, though.